testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/TestActivity.java

```diff
@@ -0,0 +1,152 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.scenarios;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import androidx.annotation.NonNull;
+import io.flutter.Log;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.loader.FlutterLoader;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.BinaryCodec;
+import io.flutter.plugin.common.JSONMethodCodec;
+import io.flutter.plugin.common.MethodChannel;
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class TestActivity extends TestableFlutterActivity {
+  static final String TAG = "Scenarios";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    final Intent launchIntent = getIntent();
+    if ("com.google.intent.action.TEST_LOOP".equals(launchIntent.getAction())) {
+      if (Build.VERSION.SDK_INT > 22) {
+        requestPermissions(new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1);
+      }
+      // Run for one minute, get the timeline data, write it, and finish.
+      final Uri logFileUri = launchIntent.getData();
+      new Handler()
+          .postDelayed(
+              new Runnable() {
+                @Override
+                public void run() {
+                  writeTimelineData(logFileUri);
+
+                  testFlutterLoaderCallbackWhenInitializedTwice();
+                }
+              },
+              20000);
+    } else {
+      testFlutterLoaderCallbackWhenInitializedTwice();
+    }
+  }
+
+  @Override
+  @NonNull
+  public FlutterShellArgs getFlutterShellArgs() {
+    FlutterShellArgs args = FlutterShellArgs.fromIntent(getIntent());
+    args.add(FlutterShellArgs.ARG_TRACE_STARTUP);
+    args.add(FlutterShellArgs.ARG_ENABLE_DART_PROFILING);
+    args.add(FlutterShellArgs.ARG_VERBOSE_LOGGING);
+    return args;
+  }
+
+  @Override
+  public void onFlutterUiDisplayed() {
+    final Intent launchIntent = getIntent();
+    if (!launchIntent.hasExtra("scenario")) {
+      return;
+    }
+    MethodChannel channel =
+        new MethodChannel(getFlutterEngine().getDartExecutor(), "driver", JSONMethodCodec.INSTANCE);
+    Map<String, Object> test = new HashMap<>(2);
+    test.put("name", launchIntent.getStringExtra("scenario"));
```

Comment:
This comes through as an integer, rather than a string, and causes the test to fail on firebase unfortunately - although we're not picking up the failure :\

Reply:
Hm - I can see this was just moved from other code - I guess FTL changed or this code just never worked.